### ISSUE TITLE
Generate impl for `T` assist should strip default type parameters

### ISSUE THREAD

**Issue by Maxpxt** [rust-analyzer/rust-analyzer#7947]
Given a generic type `Foo` with default type parameters, the ``Generate impl for `Foo` `` assist currently writes the default type parameters in the generated impl. This triggers the [`invalid_type_param_default`](https://github.com/rust-lang/rust/issues/36887) lint, which is in deny by default. The assist should strip the default type parameters.

## Example
The ``Generate impl for `Foo` `` assist on

```rust
struct Foo<T = ()>(T);
```

generates

```rust
impl<T = ()> Foo<T> {
    
}
```

It should, instead, generate

```rust
impl<T> Foo<T> {
    
}
```

**Comment by Veykril** [comment]
Relevant code is here https://github.com/rust-analyzer/rust-analyzer/blob/9a5c72d9f07760fe875ef653a956fcaa1fe5d0de/crates/ide_assists/src/utils.rs#L414-L456

**Comment by unexge** [comment]
seems like this already fixed in https://github.com/rust-analyzer/rust-analyzer/pull/7667
